napari/layers/image/_image_constants.py

```diff
@@ -88,3 +88,9 @@ class ImageRendering(StringEnum):
     MINIP = auto()
     ATTENUATED_MIP = auto()
     AVERAGE = auto()
+
+
+class ImageSliceProjection(StringEnum):
```

Comment:
I might make this called `Projection` and then have values `slice` (current default), then `average`, and `additive`. I'd also make `Projection` be on the base layer with just slice as all layers can slice. I'd then make `ImageProjection` which subclasses `Projection` and also has `average` and `additive` on it

Reply:
Yeah, that sounds much better, thanks!

Reply:
Turns out Enums cannot be subclassed... Not sure why that's the case, but it's a feature. I think we're better off doing the same as `LabelsRendering` and `ImageRendering`, where they are different classes.

Reply:
> Turns out Enums cannot be subclassed...

why do you say that?

Reply:
```python
In [1]: from enum import Enum

In [2]: class E(Enum):
   ...:     x = 1
   ...:

In [3]: class F(E):
   ...:     y = 2
   ...:
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Input In [3], in <module>
----> 1 class F(E):
      2     y = 2

File /usr/lib/python3.10/enum.py:168, in EnumMeta.__prepare__(metacls, cls, bases, **kwds)
    165 @classmethod
    166 def __prepare__(metacls, cls, bases, **kwds):
    167     # check that previous enum members do not exist
--> 168     metacls._check_for_existing_members(cls, bases)
    169     # create the namespace dict
    170     enum_dict = _EnumDict()

File /usr/lib/python3.10/enum.py:574, in EnumMeta._check_for_existing_members(class_name, bases)
    572 for base in chain.__mro__:
    573     if issubclass(base, Enum) and base._member_names_:
--> 574         raise TypeError(
    575                 "%s: cannot extend enumeration %r"
    576                 % (class_name, base.__name__)
    577                 )

TypeError: F: cannot extend enumeration 'E'
```

Reply:
ah, sorry, I was focusing on your line `class ImageSliceProjection(StringEnum):` ... yes, enums [can only have one concrete type](https://docs.python.org/3/library/enum.html#restricted-enum-subclassing) in their mro.

If you want to extend a list, you can construct the sub enum like this:
```python
F = Enum("F", {i.name:i.value for i in E.__members__.values()})
```

Reply:
I see, but this doesn't seem like it helps much at all. It's less readable, and still not a subclass.

Reply:
Another option would be to use the same enum for everything, and just whitelist some members depending on the layer type?

Reply:
yep, agreed.  the subclass won't be possible.  is there somewhere you need to it be a direct subclass? (like, check `isinstance`)

Reply:
Not really... I'll move forward with different classes for now, then see how that looks!